nvme_gw_cli.py

```diff
@@ -0,0 +1,249 @@
+            # Add specified arguments to the parser and set the function
+            # attribute to point to the subcommand's associated function
+            for arg in args:
+                parser.add_argument(*arg[0], **arg[1])
+                parser.set_defaults(func=func)
+            return func
+
+        return decorator
+
+
+class GatewayClient:
+
+    # Client for gRPC functionality with a gateway server.
+
+    # Contains methods to send RPC calls to the server and specifications for the
+    # associated command line arguments.
+
+    # Class attributes:
+    #     cli: Parser object
+
+    # Instance attributes: * Must be initialized with GatewayClient.connect *
+    #     stub: Object on which to call server methods
+    #     logger: Logger instance to track client events
+
+    cli = Parser()
+
+    def __init__(self):
+        self._stub = None
+        self._logger = None
+
+    @property
+    def stub(self):
+        # Object on which to call server methods.
+
+        if self._stub is None:
+            raise AttributeError("stub is None. Set with connect method.")
+        return self._stub
+
+    @property
+    def logger(self):
+        # Logger instance to track client events.
+
+        if self._logger is None:
+            raise AttributeError("logger is None. Set with connect method.")
+        return self._logger
+
+    def connect(self, nvme_config):
+
+        # Connects to server and sets stub and logger.
+
+        # Read in configuration parameters
+        host = nvme_config.get("config", "gateway_addr")
+        port = nvme_config.get("config", "gateway_port")
+        enable_auth = nvme_config.getboolean("config", "enable_auth")
+        server = "{}:{}".format(host, port)
+
+        if enable_auth:
+
+            # Create credentials for mutual TLS and a secure channel
+            with open(nvme_config.get("mtls", "client_cert"), "rb") as f:
+                client_cert = f.read()
+            with open(nvme_config.get("mtls", "client_key"), "rb") as f:
+                client_key = f.read()
+            with open(nvme_config.get("mtls", "server_cert"), "rb") as f:
+                server_cert = f.read()
+
+            credentials = grpc.ssl_channel_credentials(
+                root_certificates=server_cert,
+                private_key=client_key,
+                certificate_chain=client_cert,
+            )
+            channel = grpc.secure_channel(server, credentials)
+        else:
+
+            # Instantiate a channel without credentials
+            channel = grpc.insecure_channel(server)
+
+        # Bind the client and the server
+        self._stub = pb2_grpc.NVMEGatewayStub(channel)
+        # Set up logging
+        self._logger = nvme_config.logger
+
+    @cli.cmd([
+        argument("-i", "--image", help="RBD image name", required=True),
+        argument("-p", "--pool", help="Ceph pool name", required=True),
+        argument("-b", "--bdev_name", help="Bdev name"),
```

Comment:
need to be consistent with `create_namespace`
```suggestion
        argument("-b", "--bdev", help="Bdev name"),
```